contracts/simpleKitty/SimpleKittyBase.sol

```diff
@@ -0,0 +1,147 @@
+        // generation number of all other cats is the larger of the two generation
+        // numbers of their parents, plus one.
+        // (i.e. max(matron.generation, sire.generation) + 1)
+        uint16 generation;
+    }
+
+    /*** STORAGE ***/
+
+    /// @dev An array containing the Kitty struct for all Kitties in existence. The ID
+    ///  of each cat is actually an index into this array. Note that ID 0 is a negacat,
+    ///  the unKitty, the mythical beast that is the parent of all gen0 cats. A bizarre
+    ///  creature that is both matron and sire... to itself! Has an invalid genetic code.
+    ///  In other words, cat ID 0 is invalid... ;-)
+    Kitty[] internal kitties;
+
+    /// @dev A mapping from cat IDs to the address that owns them. All cats have
+    ///  some valid owner address, even gen0 cats are created with a non-zero owner.
+    mapping(uint256 => address) public kittyIndexToOwner;
+
+    // @dev A mapping from owner address to count of tokens that address owns.
+    //  Used internally inside balanceOf() to resolve ownership count.
+    mapping(address => uint256) internal ownershipTokenCount;
+
+    /// @dev A mapping from KittyIDs to an address that has been approved to call
+    ///  transferFrom(). Each Kitty can only have one approved address for transfer
+    ///  at any time. A zero value means no approval is outstanding.
+    mapping(uint256 => address) public kittyIndexToApproved;
+
+    /// @dev Assigns ownership of a specific Kitty to an address.
+    function _transfer(address _from, address _to, uint256 _tokenId) internal {
+        // Since the number of kittens is capped to 2^32 we can't overflow this
+        ownershipTokenCount[_to]++;
+        // transfer ownership
+        kittyIndexToOwner[_tokenId] = _to;
+        // When creating new kittens _from is 0x0, but we can't account that address.
+        if (_from != address(0)) {
+            ownershipTokenCount[_from]--;
+            // clear any previously approved ownership exchange
+            delete kittyIndexToApproved[_tokenId];
+        }
+        // Emit the transfer event.
+        Transfer(_from, _to, _tokenId);
+    }
+
+    /// @dev An internal method that creates a new kitty and stores it. This
+    ///  method doesn't do any checking and should only be called when the
+    ///  input data is known to be valid. Will generate both a Birth event
+    ///  and a Transfer event.
+    /// @param _matronId The kitty ID of the matron of this cat (zero for gen0)
+    /// @param _sireId The kitty ID of the sire of this cat (zero for gen0)
+    /// @param _generation The generation number of this cat, must be computed by caller.
+    /// @param _genes The kitty's genetic code.
+    /// @param _owner The inital owner of this cat, must be non-zero (except for the unKitty, ID 0)
+    function _createKitty(uint256 _matronId, uint256 _sireId, uint256 _generation, uint256 _genes, address _owner)
```

Comment:
seems that this method should work with the same set of data as `getKitty` returns:
  * cooldownIndex
  * cooldownEndBlock
  * siringWithId
  * birthTime
  * matronId
  * sireId
  * generation
  * genes

It will allow the tokens to be bridge through AMB. Moreover it should create the NFT with particular ID.

Reply:
Updated the method to receive the same information provided by `getKitty` method and updated the struct to store `isGestating` and `isReady` since they will be also provided. Also changed how the kitties are stored. Instead of using an array I replaced it with a mapping. Kitties Id is related to the array index in the original contracts, since we will have to mint/burn kitties with a specified Id a mapping makes more sense to me. Please check https://github.com/poanetwork/cryptokitties-xdai-demo/pull/5/commits/09f8684d070d4fd58f881ce7b670f798f7640330